python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -15,36 +15,49 @@
 import sys
 import textwrap
 import uuid
-import shutil
-import stat
 import tempfile
-import datetime
+import optparse
 
 from ...util import shotgun
 from ...platform import constants
-from ... import pipelineconfig
 from ... import pipelineconfig_utils
 from ..zipfilehelper import unzip_file
 from .. import util
 
 from . import console_utils
 
-        
-class CoreUpgradeAction(Action):
+
+# FIXME: This should be refactored into something that can be used by other commands.
+class TkOptParse(optparse.OptionParser):
+    def __init__(self, *args, **kwargs):
```

Comment:
Before turning this into an API feature, maybe we should look at what it means using OptionGroups and OptionParser in a more general sense in Toolkit first.

Docstrings done.